coffee-module/coffee-module-redisstream/src/main/java/hu/icellmobilsoft/coffee/module/redisstream/bootstrap/ConsumerStarterExtension.java

```diff
@@ -0,0 +1,114 @@
+ * limitations under the License.
+ * #L%
+ */
+package hu.icellmobilsoft.coffee.module.redisstream.bootstrap;
+
+import java.util.Set;
+
+import javax.enterprise.concurrent.ManagedExecutorService;
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.Instance;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.CDI;
+import javax.enterprise.inject.spi.Extension;
+import javax.naming.InitialContext;
+import javax.naming.NamingException;
+
+import hu.icellmobilsoft.coffee.dto.exception.BaseException;
+import hu.icellmobilsoft.coffee.module.redisstream.annotation.RedisStreamConsumer;
+import hu.icellmobilsoft.coffee.module.redisstream.config.StreamGroupConfig;
+import hu.icellmobilsoft.coffee.module.redisstream.consumer.IRedisStreamConsumer;
+import hu.icellmobilsoft.coffee.module.redisstream.consumer.IRedisStreamConsumerExecutor;
+import hu.icellmobilsoft.coffee.module.redisstream.consumer.RedisStreamConsumerExecutor;
+import hu.icellmobilsoft.coffee.se.logging.Logger;
+import hu.icellmobilsoft.coffee.tool.utils.annotation.AnnotationUtil;
+
+/**
+ * Create Redis stream consumer in separate threads. This way have some limits, like JNDI access. This class can be activated by
+ * <code>META-INF/services/javax.enterprise.inject.spi.Extension</code> file
+ * 
+ * @author imre.scheffer
+ * @since 1.3.0
+ *
+ */
+public class ConsumerStarterExtension implements Extension {
+
+    private static final Logger LOG = Logger.getLogger(ConsumerStarterExtension.class);
+
+    private ManagedExecutorService managedExecutorService;
+
+    /**
+     * After deployment validation
+     * 
+     * @param adv
+     *            object, not relevant
+     */
+    void validate(@Observes AfterDeploymentValidation adv, BeanManager beanManager) {
+        LOG.debug("Checking consumer for RedisStreamConsumer...");
+        // kiszedjuk az osszes olyan osztalyt, ami a IRedisStreamConsumer interfeszt implementalja
+        Set<Bean<?>> beans = beanManager.getBeans(IRedisStreamConsumer.class, RedisStreamConsumer.LITERAL);
+        LOG.info("Found [{0}] RedisStreamConsumer bean...", beans.size());
+        if (!beans.isEmpty()) {
+            Instance<StreamGroupConfig> iconfig = beanManager.createInstance().select(StreamGroupConfig.class);
+            StreamGroupConfig config = iconfig.get();
+            initManagedExecutorService(adv);
+            if (managedExecutorService == null) {
+                return;
+            }
+            beans.stream().forEach((bean) -> handleConsumerBean(bean, config));
+            iconfig.destroy(config);
+        }
+    }
+
+    private void initManagedExecutorService(AfterDeploymentValidation adv) {
+        try {
+            managedExecutorService = (ManagedExecutorService) InitialContext.doLookup("java:jboss/ee/concurrency/executor/default");
```

Comment:
valami konfigba ki lehetne tennia default-ot?

Reply:
Ezt az Extensiont csak valamifele mintanak hagytam bent, ezzel a megoldassal van mas problema is, csak nem volt ido vegig kutatni a lehetosegeket, mivel az Initialized Applicationscope fire valamiert nem mindig dobodik vagy ez nem kapja el...
Valami helyet kellene neki talani, mert nem igazan illik ebben a formaban a coffee-ba